servers/partner-ci/builders/workstation.sh

```diff
@@ -94,26 +94,44 @@ print data
 client.close()
 CONTENT
 
-chmod +x $SSH_ENDPOINT
+sudo chmod +x $SSH_ENDPOINT
 
 }
 
+esxi_exec(){
+
+  set -x
+  host=$1
+  cmd=$2
+  echo "$cmd-------------------------------------------------------"
+  python2 $SSH_ENDPOINT $host $ESXI_USER $ESXI_PASSWORD "$cmd"
+
+}
 configure_nfs(){
   set -x
   create_ssh_endpoint
-  source $VENV_PATH/bin/activate
   for esxi_host in $ESXI_HOSTS; do
-
-    python2 $SSH_ENDPOINT $esxi_host $ESXI_USER $ESXI_PASSWORD 'storages=$(esxcli storage nfs list | grep nfs | cut -d" " -f1); for storage in $storages; do esxcli storage nfs remove -v $storage; done'
-    echo "nfs storages have been successfully removed for $esxi_host"
+    esxi_exec $esxi_host './sbin/services.sh restart'
+    esxi_exec $esxi_host '/etc/init.d/sfcbd-watchdog stop; rm -rf /var/run/sfcb/*;/etc/init.d/sfcbd-watchdog start'
+    esxi_exec $esxi_host 'esxcli network firewall set --enabled false'
+    esxi_exec $esxi_host 'esxcli system hostname get'
+    echo "++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++"
+    if [[ ${DEBUG} == "true" ]]; then
+      esxi_exec $esxi_host 'esxcli storage nfs list'
+      echo "----------------------------------------------------------------------------------------------------"
+    fi
+    esxi_exec $esxi_host 'storages=$(esxcli storage nfs list | grep nfs | cut -d" " -f1); for storage in $storages; do esxcli storage nfs remove -v $storage; done'
 
     for nfs_share in $NFS_SHARES; do
-      python2 $SSH_ENDPOINT $esxi_host $ESXI_USER $ESXI_PASSWORD "esxcli storage nfs add -H $NFS_SERVER -s /var/$nfs_share -v $nfs_share"
-      echo "$nfs_share has been successfully connected for $esxi_host"
+      if esxi_exec $esxi_host "esxcli storage nfs add -H $NFS_SERVER -s /var/$nfs_share -v $nfs_share"; then
```

Comment:
- do nfs_share addition on esxi host `esxi_exec "for nfs_share in $NFS_SHARES"`...
- create a variable with list of commands and then execute them in cycle
- may be you need to reassign an error-code from paramiko 
- after 'esxcli storage core adapter rescan --all' need to check, that nfs are connected and active 
OR 
you can use a simple ansible-playbook to do all that stuff